File: src/main/java/com/procurement/access/model/dto/lots/LotDto.java
package com.procurement.access.model.dto.lots;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;

@Getter
public class LotDto {

    @JsonProperty(value = "id")
    private String id;

    public LotDto(@JsonProperty("id") final String id) {
        this.id = id;
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder()
                .append(id)
                .toHashCode();
    }

    @Override
    public boolean equals(final Object other) {
        if (other == this) {
            return true;
        }
        if (!(other instanceof LotDto)) {
            return false;
        }
        final LotDto rhs = (LotDto) other;
        return new EqualsBuilder()
                .append(id, rhs.id)
                .isEquals();
    }
}

File: src/main/kotlin/com/procurement/access/model/dto/lots/LotDto.kt
package com.procurement.access.model.dto.lots

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.procurement.access.model.dto.ocds.Value

@JsonInclude(JsonInclude.Include.NON_NULL)
data class LotDto @JsonCreator constructor(

        val id: String?,

        var title: String?,

        var description: String?,

        val value: Value?
)